clusterloader2/pkg/measurement/common/system_pod_metris.go

```
@@ -0,0 +1,127 @@
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	systemPodMetricsName            = "SystemPodMetrics"
+	systemNamespace                 = "kube-system"
+	systemPodMetricsEnabledFlagName = "systemPodMetricsEnabled"
+)
+
+func init() {
+	if err := measurement.Register(systemPodMetricsName, createSystemPodMetricsMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", systemPodMetricsName, err)
+	}
+}
+
+func createSystemPodMetricsMeasurement() measurement.Measurement {
+	return &systemPodMetricsMeasurement{}
+}
+
+type systemPodMetricsMeasurement struct{}
+
+type containerMetrics struct {
+	Name         string `json:"name"`
+	RestartCount int32  `json:"restartCount"`
+}
+
+type podMetrics struct {
+	Name       string             `json:"name"`
+	Containers []containerMetrics `json:"containers"`
+}
+
+type systemPodsMetrics struct {
+	Pods []podMetrics `json:"pods"`
+}
+
+// Execute gathers and prints system pod metrics.
+func (m *systemPodMetricsMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	if !systemPodMetricsEnabled(config) {
+		klog.Info("skipping collection of system pod metrics")
+		return []measurement.Summary{}, nil
+	}
+
+	klog.Info("collecting system pod metrics...")
+	lst, err := getPodList(config.ClusterFramework.GetClientSets().GetClient())
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := util.PrettyPrintJSON(extractMetrics(lst))
+	if err != nil {
+		return nil, err
+	}
+
+	summary := measurement.CreateSummary(systemPodMetricsName, "json", content)
+	return []measurement.Summary{summary}, nil
+}
+
+func systemPodMetricsEnabled(config *measurement.MeasurementConfig) bool {
+	systemPodMetricsEnabled, err := util.GetBoolOrDefault(config.Params, systemPodMetricsEnabledFlagName, false)
+	if err != nil {
+		klog.Errorf("failed to check %v flag: %v", systemPodMetricsEnabledFlagName, err)
+		return false
+	}
```

I'd rather not swallow the error and would return it. Fail fast!

---

makes sense